astropy/cosmology/_io/model.py

```diff
@@ -95,12 +99,13 @@ def cosmology(self):
         return cosmo
 
     @classproperty
-    def method_name(self):
+    def method_name(self) -> str:
         """Redshift-method name on |Cosmology| instance."""
         return self._method_name
 
     # ---------------------------------------------------------------
 
+    # NOTE: cannot add type annotations b/c of how Model introspects
```

Comment:
Are you implying that `inspect.isabstract` doesn't play nice with type annotations or do you mean something else ?

Reply:
Astropy Model introspects the type annotations of the `evaluate` method to figure out units. It breaks with these non-unitful annotations.

Reply:
That sounds like it could be a problem for users. But fixing that is beyond the scope here.

Reply:
Agreed.